lib/core/src/system/web/text_input/text_input.js

```diff
@@ -0,0 +1,128 @@
+// Text input event handlers.
+//
+// The "text input event" is the normal keyup/keydown event, or cut/copy/paste event. The handling of these events is
+// done by creating an invisible focused textarea html element and intercepting events emitted to this element.
+// We do it this way, because this is so far the only way a website may affect the clipboard which works on each
+// browser.
+export class TextInputHandlers {
+    // Constructor creates the textarea element and put it into document body.
+    constructor() {
+        this.event_handlers = {}
+        this.text_area = create_invisible_text_area()
+        document.body.appendChild(this.text_area)
+        this.text_area.focus()
+        this.bind_text_area_events()
+    }
+
+    // Set event handler. The name can be 'keyup' or 'keydown'.
+    set_event_handler(name, callback) { this.event_handlers[name] = callback }
+    // Set copy handler. The copy handler takes bool as first argument denoting if it is cut operation, and returns
+    // string which should be actually copied to clipboard.
+    set_copy_handler(handler)         { this.copy_handler          = handler }
+    // Set paste handler. The paste handler takes the text from clipboard as the only argument.
+    set_paste_handler(handler)        { this.paste_handler         = handler }
+
+    // Remove the textarea element and stop handling any events.
+    stop_handling() {
+        this.text_area.remove()
+    }
+
+    // This is private function being a construction stage.
+    bind_text_area_events() {
+        var ta = this.text_area
+
+        ta.addEventListener('cut', e => {
+            setTimeout(_ => {ta.value = "";}, 0)
```

Comment:
What this is for? What this `setTimeout` does here?

Reply:
It's just "clear the text before next frame". I'll add comment for this